addons/portal_rating/static/src/chatter/frontend/composer_patch.js

```diff
@@ -0,0 +1,54 @@
+import { Composer } from "@mail/core/common/composer";
+
+import { patch } from "@web/core/utils/patch";
+import { rpc } from "@web/core/network/rpc";
+import { useState } from "@odoo/owl";
+
+patch(Composer.prototype, {
+    setup() {
+        super.setup(...arguments);
+        this.state = useState({
+            ...this.state,
```

Comment:
I'm not sure if combining state works very well.

I'd suggest to have a separate "portal state" instead.